File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        HashMap<Integer,Integer> h = new HashMap<>();
        ArrayList<Integer> res = new ArrayList<>();

        for(Integer i : nums){
            if(h.containsKey(i)){
                res.add(i);
            }else{
                h.put(i,1);
            }
        }
        return res;
    }
}

File: 442-find-all-duplicates-in-an-array/find-all-duplicates-in-an-array.dart
class Solution {
  List<int> findDuplicates(List<int> nums) {
 final Set<int> seen = {};
    final Set<int> duplicates = {}; 
    
    for (final i in nums) {
      if (seen.contains(i)) {
        duplicates.add(i);
      } else {
        seen.add(i);
      }
    }
    
    return duplicates.toList();
  }
}